src/mcp-sdk/tests/Server/RequestHandler/PromptListHandlerTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\McpSdk\Tests\Server\RequestHandler;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Small;
+use PHPUnit\Framework\TestCase;
+use Symfony\AI\McpSdk\Capability\Prompt\CollectionInterface;
+use Symfony\AI\McpSdk\Capability\Prompt\MetadataInterface;
+use Symfony\AI\McpSdk\Message\Request;
+use Symfony\AI\McpSdk\Server\RequestHandler\PromptListHandler;
+
+#[Small]
+#[CoversClass(PromptListHandler::class)]
+class PromptListHandlerTest extends TestCase
+{
+    public function testHandleEmpty(): void
+    {
+        $collection = $this->getMockBuilder(CollectionInterface::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['getMetadata'])
+            ->getMock();
+        $collection->expects($this->once())->method('getMetadata')->willReturn([]);
```

Comment:
For me it would be totally fine to go with a real instance instead:
```suggestion
        $collection = new PromptChain([]);
```